src/client/dfs/dfs.c

```diff
@@ -3264,7 +3267,7 @@ dfs_osetattr(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags)
 	}
 
 out_stat:
-	rc = entry_stat(dfs, th, oh, obj->name, strlen(obj->name), stbuf);
+	rc = entry_stat(dfs, th, oh, obj->name, strlen(obj->name), obj, stbuf);
```

Comment:
im not sure i understand what you are trying to check for here.
If the object entry is wrong (someone changed it), you already updated the wrong entry in the update; and you are just doing the check after the fact?

Reply:
I was going to write another test for that, and fix this at the same time.  I can roll it onto this PR though.

Reply:
I think the fix for this is going to be a bit more involved.

Reply:
Done.  I added a check that no attempt is made to change the type of a file as well, although dfs_osetattr() is still not using transactions.